apps/namadillo/src/atoms/ledger/atoms.ts

```diff
@@ -0,0 +1,67 @@
+import { ledgerUSBList } from "@namada/sdk/web";
+import { LedgerError } from "@zondax/ledger-namada";
+import { isLedgerAccountAtom } from "atoms/accounts";
+import { atom } from "jotai";
+
+import { atomWithQuery } from "jotai-tanstack-query";
+import { getSdkInstance } from "utils/sdk";
+
+export type LedgerStatus = {
+  connected: boolean;
+  errorMessage: string;
+};
+
+export const ledgerStatusAtom = atomWithQuery<LedgerStatus | undefined>(() => {
+  return {
+    refetchInterval: 1000,
+    queryKey: ["ledger-status"],
+    queryFn: async () => {
```

Comment:
I think something may need to change here after testing. What I found was that this loop continues to execute _after_ I've connected and submitted a Tx, and I hit a few cases with different errors thrown. It will definitely be an issue if we try to open transport on Web when the Keychain is attempting to open transport to sign. I'll give it some thought! Maybe if we, say, detect it once, if it's undetected, require the user to click to _Connect_, once connected, fetch build params then submit. I think maybe it would be better if validation happens once, or, if the check is discontinued after first successful connection? Not sure, will take some thought